orderer/consensus/etcdraft/blockpuller.go

```diff
@@ -108,3 +120,109 @@ func NewBlockPuller(support consensus.ConsenterSupport,
 		BlockPuller:    bp,
 	}, nil
 }
+
+// NewBlockFetcher creates a new block fetcher
+func NewBlockFetcher(support consensus.ConsenterSupport,
+	baseDialer *cluster.PredicateDialer,
+	clusterConfig localconfig.Cluster,
+	bccsp bccsp.BCCSP,
+) (BlockPuller, error) {
+	verifyBlockSequence := func(blocks []*common.Block, _ string) error {
+		return cluster.VerifyBlocks(blocks, support)
+	}
+
+	stdDialer := &replication.StandardDialer{
+		Config: baseDialer.Config,
+	}
+	stdDialer.Config.AsyncConnect = false
+	stdDialer.Config.SecOpts.VerifyCertificate = nil
+
+	// Extract the TLS CA certs and endpoints from the configuration,
+	endpoints, err := EndpointconfigFromSupport(support, bccsp)
+	if err != nil {
+		return nil, err
+	}
+
+	der, _ := pem.Decode(stdDialer.Config.SecOpts.Certificate)
+	if der == nil {
+		return nil, errors.Errorf("client certificate isn't in PEM format: %v",
+			string(stdDialer.Config.SecOpts.Certificate))
+	}
+
+	fc := replication.FetcherConfig{
+		Channel:      support.ChannelID(),
+		Signer:       support,
+		TLSCert:      der.Bytes,
+		Dialer:       stdDialer,
+		Endpoints:    endpoints,
+		FetchTimeout: clusterConfig.ReplicationPullTimeout,
+	}
+
+	// To tolerate byzantine behaviour of `f` faulty nodes, we need a total of `3f + 1` nodes.
+	// f = maxByzantineNodes, total = len(endpoints)
+	maxByzantineNodes := uint64(len(endpoints)-1) / 3
+
+	// timeout for time based shuffling
+	// 10 times the fetchtimeout
+	shuffleTimeout := shuffleTimeoutMultiplier * clusterConfig.ReplicationPullTimeout
+	shuffleTimeoutThrehold := shuffleTimeoutPercentage
+
+	bf := replication.BlockFetcher{
```

Comment:
where are you taking this code from? 

The `main2` branch doesn't have this:

https://github.com/Fabric-orderer-collaboration/fabric/blob/main2/orderer/consensus/etcdraft/blockpuller.go